### PR TITLE
Updates to API 404 responses

### PR DESCRIPTION
**Pull request by jolheiser**
This PR will address #6052  
  
So far it handles API 404s.  
Instead of a blank 404, it returns a JSON response with a `message` of "Not Found" and a `documentation_url` pointing to the basic swagger endpoint. e.g. `https://try.gitea.io/api/swagger`

### CODE REVIEW

**Review by zeripath — COMMENTED** [review]
I'd prefer a path.Join in there instead of fmt.Sprintf

**Comment by lafriks** [comment]
It should probably moved to some kind of helper method as all 404 errors (also ones returned from methods where record by id is not found etc) should return same imho

**Review by BetaCat0 — APPROVED** [review]

File: routers/api/v1/api.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ import (
 	api "code.gitea.io/sdk/gitea"
 
 	"github.com/go-macaron/binding"
-	macaron "gopkg.in/macaron.v1"
```
Comment (jolheiser) [inline routers/api/v1/api.go:77]: This must have happened automatically via GoLand. Let me know if you want me to revert it, but it must work without it.

File: modules/context/api.go [hunk 2]
```diff
@@ -140,3 +143,28 @@ func ReferencesGitRepo() macaron.Handler {
...
+		} else {
+			message = obj.(string)
+		}
+	}
+
+	swagger, err := url.Parse(setting.AppURL)
+	if err != nil {
+		ctx.Error(500, "Invalid AppURL", err)
+	}
+	swagger.Path = path.Join(swagger.Path, "api", "swagger")
```
Comment (lafriks) [inline modules/context/api.go]: Why not just use this?
```suggestion
	swagger.Path = path.Join(setting.AppURL, "api", "swagger")
```

File: modules/context/api.go [hunk 3]
```diff
@@ -140,3 +143,28 @@ func ReferencesGitRepo() macaron.Handler {
...
+	var errors []string
+	for _, obj := range objs {
+		if err, ok := obj.(error); ok {
+			errors = append(errors, err.Error())
+		} else {
+			message = obj.(string)
+		}
+	}
+
+	swagger, err := url.Parse(setting.AppURL)
```
Comment (jolheiser) [inline modules/context/api.go]: Without parsing the URL, the `path.Join` call was replacing the double slash in `http(s)://` with a single slash

File: modules/context/api.go (hunk 2 above)
Comment (jolheiser) [reply modules/context/api.go]: Without parsing the URL, the `path.Join` call was replacing the double slash in `http(s)://` with a single slash

Comment (lafriks) [reply modules/context/api.go]: yeah right, now I remember... Only thing may be rename `swagger` variable name to something like `u` or `appURL`

File: modules/context/api.go [hunk 4]
```diff
@@ -140,3 +143,28 @@ func ReferencesGitRepo() macaron.Handler {
...
+		if err, ok := obj.(error); ok {
+			errors = append(errors, err.Error())
+		} else {
+			message = obj.(string)
+		}
+	}
+
+	u, err := url.Parse(setting.AppURL)
+	if err != nil {
+		ctx.Error(500, "Invalid AppURL", err)
```
Comment (adelowo) [inline modules/context/api.go:162]: Shouldn't there be a return here?

Comment (jolheiser) [reply modules/context/api.go:162]: Good catch. 😅  
Still getting used to using the context.

**Review by adelowo — APPROVED** [review]

**Comment by jolheiser** [comment]
For the record, I had planned on fixing both things in the original issue, but perhaps this should be a PR on its own just for API 404s. 😄

**Comment by techknowlogick** [comment]
This PR is marked as WIP, is it still WIP?

**Comment by jolheiser** [comment]
I guess it depends on whether we want the other sub-issue in the original issue fixed in this PR or a separate one. If not, this is complete for API 404s.

**Comment by jolheiser** [comment]
No comment on the original issue, so I will count this PR as just fixes to API 404s.

**Review by zeripath — APPROVED** [review]

**Pull request merged**